Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2606,7 +2623,7 @@ extension BrowserViewController: TopSitesDelegate {
 extension BrowserViewController: PreferencesObserver {
     func preferencesDidChange(for key: String) {
         HTTPCookieStorage.shared.updateCookieAcceptPolicy(to: HTTPCookie.AcceptPolicy(rawValue: Preferences.Privacy.cookieAcceptPolicy.value))
-        // TODO: Update tab bar visiblity based on `Preferences.tabBarVisibility` once BraveURLBarView is back
+        updateTabsBarVisibility()
```

Comment:
Shouldn't we utilize `key` param here to update tab bar visibility only if tabs bar related preferences are changed? Now it triggers on every preference change. Same goes with HTTP cookie storage line above.

So every shields popup tick would call all this code, maybe I'm missing something.

Reply:
Good call, made me realize I didn't even setup observing for the `Preferences.General.tabBarVisibility` preference at all.

Updated so it only does relevant actions based on key